src/rust/api/src/routes.rs

```diff
@@ -0,0 +1,114 @@
+use axum::{extract::State, Json};
+use sqlx::{Pool, Postgres};
+use types::error::TypeError;
+
+use crate::error::ApiError;
+
+pub async fn index() -> String {
+    String::from("Econia backend API")
+}
+
+pub async fn markets(
+    State(pool): State<Pool<Postgres>>,
+) -> Result<Json<Vec<types::Market>>, ApiError> {
+    let query_markets = sqlx::query_as!(
+        types::QueryMarket,
+        r#"
+        select 
+            market_id,
+            base_coins.symbol as base_symbol,
+            base_coins.name as base_name,
+            base_coins.decimals as base_decimals,
+            base_coins.account_address as base_account_address,
+            base_coins.module_name as base_module_name,
+            base_coins.struct_name as base_struct_name,
```

Comment:
This would be more appropriate as `base_asset`, since the base asset can be either a coin or a `GenericAsset`.